src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.Calculator/NumberTranslator.cs

```diff
@@ -76,10 +76,30 @@ private static string Translate(string input, CultureInfo cultureFrom, CultureIn
             string[] tokens = splitRegex.Split(input);
             foreach (string token in tokens)
             {
+                int leadingZeroCount = 0;
+
+                // Count leading zero characters.
+                foreach (char c in token)
+                {
+                    if (c != '0')
+                    {
+                        break;
+                    }
+
+                    leadingZeroCount++;
+                }
+
+                // number is all zero characters. no need to add zero characters at the end.
```

Comment:
- Does that mean an input like `50.00` comes out as `50.`, maybe? (With an unnecessary decimal character hanging)
- And am I correct if I see that this applies to both the integer and the fractional part? If so, doesn't that mean `0.0` (=number) becomes `.` (NaN) ?

Reply:
We don't add 0s at the end because the output of number.ToString() will be 0 already.